src/build.py

```diff
@@ -343,6 +344,8 @@ def GitCloneFetchCheckout(self):
     """Clone a git repo if not already cloned, then fetch and checkout."""
     if os.path.isdir(self.src_dir):
       print '%s directory already exists' % self.name
+      proc.check_call(['git', 'remote', 'set-url', 'origin', self.git_repo],
```

Comment:
I like that idea. I think `git clone` will create the `origin` remote by default, but we can also add a `waterfall` remote that can duplicate it (or not, if the user changes `origin`). Then in the places where we hardcode `origin/master` as the revision we check out now, we can change those to `waterfall`.